src/tutorials/0004-mutable-file-system/10.md

```diff
@@ -8,23 +8,29 @@ MFS has a [`files.read`](https://github.com/ipfs/interface-js-ipfs-core/blob/mas
 The method takes this format:
 
 ```js
-await ipfs.files.read(path, [options])
+ipfs.files.read(path, [options])
 ```
 
 The `path` provided is the path of the file to read, and it must point to a file rather than a directory.
 
-The `files.read` method returns a Buffer, which can be converted to a string using the method `toString('utf8')`. For example:
+The `files.read` method returns an Async Iterable, that iterates over the file's chunks of data, i.e. Buffers, which can be converted to a string using the method `toString('utf8')`. For example:
 
 ```js
-let bufferedContents = await ipfs.files.read('/directory/some-file.txt')  // a buffer
+const concat = require('it-concat')
+
+let bufferedContents = await concat(ipfs.files.read('/directory/some-file.txt'))  // a buffer
 let contents = bufferedContents.toString('utf8') // a string
 ```
 
 or
 
 ```js
-let contents = (await ipfs.files.read('/directory/some-file.txt')).toString('utf8') // a string
+const concat = require('it-concat')
+
+let contents = (await concat(ipfs.files.read('/directory/some-file.txt'))).toString('utf8') // a string
 // notice the parentheses around the entire await statement
 ```
 
-When you're ready to try this in the real world, you should note that `files.read` method can result in heavy memory usage based on the contents of the file being read. You might want to explore the [`files.readReadableStream`](https://github.com/ipfs/interface-js-ipfs-core/blob/master/SPEC/FILES.md#filesreadreadablestream) or [`files.readPullStream`](https://github.com/ipfs/interface-js-ipfs-core/blob/master/SPEC/FILES.md#filesreadpullstream) methods instead.
+When you're ready to try this in the real world, you should note that the above example can result in heavy memory usage, depending on the contents of the file being read. If you find this to be the case, you might want to skip the usage of the `it-concat` package and process each chunk of data iteratively.
+This is the main reason why ipfs returns `Async Iterables` so that we can decide if we want to be aware of memory performance issues or not.
+But in ProtoSchool tutorials, we don't really need to worry about this since our code challenges will use small files, so we can concatenate everything without worrying about performance.
```

Comment:
```suggestion
In ProtoSchool tutorials, we don't really need to worry about this since our code challenges use small files, so we can concatenate everything without worrying about performance.
```